x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/AbstractFilterBlock.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import java.util.Arrays;
+
+abstract class AbstractFilterBlock<B extends Block> extends AbstractBlock implements Block {
+
+    // TODO: positions need to be tracked and released from the breaker on closeInternal
+    protected final int[] positions;
+    protected final B block;
+
+    AbstractFilterBlock(B block, int[] positions) {
+        // TODO: assert positions here, also check that we map to valid positions
+        super(positions.length, block.blockFactory());
+        this.positions = positions;
+        this.block = block;
+    }
+
+    @Override
+    public ElementType elementType() {
+        return block.elementType();
+    }
+
+    @Override
+    public boolean isNull(int position) {
+        return block.isNull(mapPosition(position));
+    }
+
+    @Override
+    public boolean mayHaveNulls() {
+        return block.mayHaveNulls();
+    }
+
+    @Override
+    public boolean areAllValuesNull() {
+        return block.areAllValuesNull();
+    }
+
+    @Override
+    public boolean mayHaveMultivaluedFields() {
+        /*
+         * This could return a false positive. The block may have multivalued
+         * fields, but we're not pointing to any of them. That's acceptable.
+         */
+        return block.mayHaveMultivaluedFields();
+    }
+
+    @Override
+    public final int nullValuesCount() {
+        if (mayHaveNulls() == false) {
+            return 0;
+        } else if (areAllValuesNull()) {
+            return getPositionCount();
+        } else {
+            int nulls = 0;
+            for (int i = 0; i < getPositionCount(); i++) {
+                if (isNull(i)) {
+                    nulls++;
+                }
+            }
+            return nulls;
+        }
+    }
+
+    @Override
+    public final int getTotalValueCount() {
+        if (positions.length == block.getPositionCount()) {
```

Comment:
Hmm.............. I don't know that this is worth it. Maybe it should be the caller's responsibility not to construct a `FilteredBlock` in the case when the filter is "keep everything". And if you do make a `FilteredBlock` here then it's just slower to use.